mozilla-release/browser/base/content/tabbrowser.xml

```diff
@@ -6592,6 +6591,12 @@
         }
       ]]>
       </handler>
+      <handler event="TabRemotenessChange">
```

Comment:
Yes, a new TabParent is created when a remote <xul:browser> is bound to the DOM.

There are other events you could listen for for the other case - EndSwapDocShells, for example: http://searchfox.org/mozilla-central/rev/f5c9e9a249637c9abd88754c8963ecb3838475cb/toolkit/content/widgets/browser.xml#1354